pkg/zonemgr/zone_file.go

```diff
@@ -19,3 +25,42 @@ func NewZoneFile(fileName string) *ZoneFile {
 func (zoneFile *ZoneFile) writeFile(content string) (err error) {
 	return os.WriteFile(zoneFile.zoneFileFullName, []byte(content), zoneFilePerm)
 }
+
+func (zoneFile *ZoneFile) readFile() ([]byte, error) {
+	return os.ReadFile(zoneFile.zoneFileFullName)
+}
+
+func (zoneFile *ZoneFile) isFileExist() (*bool, error) {
+	var err error
+	isExist := false
+	if _, err = os.Stat(zoneFile.zoneFileFullName); err == nil {
+		isExist = true
+	} else if errors.Is(err, os.ErrNotExist) {
+		err = nil
+	}
+	return &isExist, err
+}
+
+func (zoneFile *ZoneFile) ReadSoaSerial() (*int, error) {
+	if isFileExist, err := zoneFile.isFileExist(); !*isFileExist || err != nil {
+		return nil, err
+	}
+	if content, err := zoneFile.readFile(); content == nil || err != nil {
+		return nil, err
+	} else {
+		return fetchSoaSerial(string(content))
+	}
+}
+
+func fetchSoaSerial(content string) (*int, error) {
```

Comment:
why we need *int not just int ?

please use uint32

soaSerialInt can be named soaSerial

Reply:
since I have to be able to return nil

Reply:
just return 0, why we need nil ?

Reply:
https://github.com/kubevirt/kubesecondarydns/pull/45#discussion_r1054124649